### PR TITLE
Rework npy copy to integrate with query processor pipeline

### PR DESCRIPTION
**Pull request by aziz-mu**
This PR implements https://github.com/kuzudb/kuzu/issues/1670 , by removing classes specific to NPY reading that aren't needed anymore, implementing a read_npy operator to match the read_csv and read_parquet operators, and making changes to CopyNode so that copying can still be done column-by-column

### CODE REVIEW

**Comment by aziz-mu** [comment]
Note that there's still a bug in the PR - reading large (>2048 rows), multidimensional (e.g. column w/ type INT32[10]) .npy files causes an error. I'm currently working on fixing this, but if it's urgent to get this PR merged to integrate with storage changes for the next release, I propose removing the failing test (which I've done already), and creating an issue for it to be fixed soon. Happy to discuss this further

File: src/storage/copier/npy_reader.cpp [hunk 1]
```diff
@@ -224,10 +224,12 @@ std::shared_ptr<arrow::DataType> NpyReader::getArrowType() const {
 std::shared_ptr<arrow::RecordBatch> NpyReader::readBlock(common::block_idx_t blockIdx) const {
```
Comment (ray6080) [inline src/storage/copier/npy_reader.cpp:224]: The returned result is incorrect for fixed-sized list. Arrow's fixed-sized list requires special construction. It should be something similar to this:
```
    auto buffer = std::make_shared<arrow::Buffer>(rowPointer, CopyConstants::NUM_ROWS_PER_BLOCK_FOR_NPY);
    int64_t length = std::min(CopyConstants::NUM_ROWS_PER_BLOCK_FOR_NPY, getNumRows() - rowNumber);
    std::shared_ptr<arrow::Field> field;
    std::shared_ptr<arrow::Array> arr;
    if (getNumDimensions() > 1) {
        auto elementField = std::make_shared<arrow::Field>(defaultFieldName, arrowType);
        auto fixedListArrowType = arrow::fixed_size_list(elementField, (int32_t)length);
        field = std::make_shared<arrow::Field>(defaultFieldName, fixedListArrowType);
        auto valuesArr = std::make_shared<arrow::PrimitiveArray>(arrowType, length * getNumElementsPerRow(), buffer);
        arr = arrow::FixedSizeListArray::FromArrays(valuesArr, (int32_t)getNumElementsPerRow()).ValueOrDie();
    } else {
        field = std::make_shared<arrow::Field>(defaultFieldName, arrowType);
        arr = std::make_shared<arrow::PrimitiveArray>(arrowType, length, buffer);
    }
```

File: src/processor/operator/copy/read_npy.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        std::unique_ptr<storage::NpyReader> reader = make_unique<storage::NpyReader>(filePath);
+        numRows = reader->getNumRows();
+        if (idx == 0) {
+            firstFileRows = numRows;
+        }
+        auto tableType = tableSchema->getProperty(idx).dataType;
+        reader->validate(tableType, firstFileRows, tableSchema->tableName);
+        auto numBlocks = (uint64_t)((numRows + common::CopyConstants::NUM_ROWS_PER_BLOCK_FOR_NPY) /
+                                    common::CopyConstants::NUM_ROWS_PER_BLOCK_FOR_NPY);
+        std::vector<uint64_t> numRowsPerBlock(numRows,
```
Comment (ray6080) [inline src/processor/operator/copy/read_npy.cpp:22]: This line should be incorrect.
It should be `std::vector<uint64_t> numRowsPerBlock(numBlocks, common::CopyConstants::NUM_ROWS_PER_BLOCK_FOR_NPY)`. Or 
```
        auto blockIdx = 0;
        std::vector<uint64_t> numRowsPerBlock(numBlocks);
        while (blockIdx < numBlocks) {
            numRowsPerBlock[blockIdx] = std::min(CopyConstants::NUM_ROWS_PER_BLOCK_FOR_NPY,
                numRows - (blockIdx * CopyConstants::NUM_ROWS_PER_BLOCK_FOR_NPY));
            blockIdx++;
        }
```

**Review by ray6080 — CHANGES_REQUESTED** [review]
There still seems to be some other bugs. I will take a look too.

**Review by ray6080 — APPROVED** [review]

**Pull request merged**